### PR TITLE
Ensure audit logging of tsh app login.

### PR DESCRIPTION
**Pull request by mdwn**
Application sessions were previously only logged when launching an application
session via the UI, and not from the `tsh app login` command. This has been
corrected. The AppName and AppURI are now passed in as part of the gRPC
request to the auth server, which is then used to emit the audit event.

Addresses https://github.com/gravitational/teleport/issues/9536

### CODE REVIEW

**Review by probakowski — APPROVED** [review]

File: lib/auth/sessions.go [hunk 1]
```diff
@@ -103,6 +109,49 @@ func (s *Server) CreateAppSession(ctx context.Context, req types.CreateAppSessio
...
+
+	// Record peer for the address of the requesting connection.
+	p, ok := peer.FromContext(ctx)
+	if !ok {
+		return nil, trace.Wrap(errors.New("unable to get peer from context"))
+	}
+
+	// Now that the certificate has been issued, emit a "new session created"
+	// for all events associated with this certificate.
+	appSessionStartEvent := &apievents.AppSessionStart{
```
Comment (ravicious) [inline lib/auth/sessions.go:126]: It looks like something might be slightly off. I rebuilt tsh and teleport, restarted my cluster and then logged in again, then proceed to login to my app through tsh. In the audit log I saw "User [rav] has connected to application [undefined]".

I'm not sure if something is wrong with the code or my setup but logging in through the Web UI produces an event with a correct app name.

Comment (ravicious) [inline lib/auth/sessions.go:126]: Also, whenever I log in to an app through tsh, I get two "User certificate issued for [rav]" events. I'm also not sure if that's the way it's supposed to work. In any case, that's probably beyond the scope of this PR as it happens on master too.

Comment (mdwn) [reply lib/auth/sessions.go:126]: Thank you, good catch. I'll check this out.

Comment (ravicious) [reply lib/auth/sessions.go:126]: Ah, sorry, I forgot that I installed the tsh.app from our downloads page a couple of weeks ago and it overshadowed my usual setup that was to use `tsh` from the build folder.

I just tested it with tsh from your branch and everything works as expected. 👍

**Review by ravicious — APPROVED** [review]

File: lib/auth/sessions.go (hunk 1 above)
Comment (mdwn) [reply lib/auth/sessions.go:126]: Awesome, thank you! :-) I appreciate it!

**Pull request merged**